docs/book/src/cronjob-tutorial/testdata/project/Makefile

```diff
@@ -114,21 +114,19 @@ docker-push: ## Push docker image with the manager.
 	$(CONTAINER_TOOL) push ${IMG}
 
 # PLATFORMS defines the target platforms for the manager image be built to provide support to multiple
-# architectures. (i.e. make docker-buildx IMG=myregistry/mypoperator:0.0.1). To use this option you need to:
+# architectures.
+PLATFORMS ?= linux/arm64,linux/amd64,linux/s390x,linux/ppc64le
+# To use this option you need to:
 # - be able to use docker buildx. More info: https://docs.docker.com/build/buildx/
 # - have enabled BuildKit. More info: https://docs.docker.com/develop/develop-images/build_enhancements/
 # - be able to push the image to your registry (i.e. if you do not set a valid value via IMG=<myregistry/image:<tag>> then the export will fail)
 # To adequately provide solutions that are compatible with multiple platforms, you should consider using this option.
-PLATFORMS ?= linux/arm64,linux/amd64,linux/s390x,linux/ppc64le
 .PHONY: docker-buildx
 docker-buildx: ## Build and push docker image for the manager for cross-platform support
-	# copy existing Dockerfile and insert --platform=${BUILDPLATFORM} into Dockerfile.cross, and preserve the original Dockerfile
-	sed -e '1 s/\(^FROM\)/FROM --platform=\$$\{BUILDPLATFORM\}/; t' -e ' 1,// s//FROM --platform=\$$\{BUILDPLATFORM\}/' Dockerfile > Dockerfile.cross
 	- $(CONTAINER_TOOL) buildx create --name project-builder
 	$(CONTAINER_TOOL) buildx use project-builder
-	- $(CONTAINER_TOOL) buildx build --push --platform=$(PLATFORMS) --tag ${IMG} -f Dockerfile.cross .
+	- $(CONTAINER_TOOL) buildx build --push --platform=$(PLATFORMS) --tag ${IMG} .
 	- $(CONTAINER_TOOL) buildx rm project-builder
```

Comment:
sure, CI approach is good, will see how to create a test for it

Reply:
I just want to share with you that it is implemented as it is because, as far as I remember, the suggestion you are doing here did not work well.

Reply:
created separate PR for CI https://github.com/kubernetes-sigs/kubebuilder/pull/4381

my idea is merge that one first (ensure buildx commands works in master branch), then check that this change does not break the job

what you think?

Reply:
Did you run it manually?
Did you check that the final image has all platforms?
If so can you please add in description?

Reply:
I can test (my primary machine does not have docker to test, but I have another one that has it) and post results

Reply:
I found this https://github.com/hadolint/hadolint/wiki/DL3029 that would also go against these changes :grimacing:

Reply:
So, I think we do not need to worry about `make docker-buildx` command with this change. Basically this changes `Dockerfile` to be equal to the `Dockerfile.cross` that would be created by the command

Reply:
I’m not sure if I fully understand what you mean, but I believe we should avoid changing this code unless we are 100% certain the change will bring value and address all requirements. When this code was originally written, it was reviewed by three experts with extensive experience in this area, and they identified and addressed many issues and scenarios.

It’s hard to recall all the details now since it was implemented over two years ago. I’m open to any changes or improvements that add value, but I’m concerned that, in this case, we might inadvertently introduce an issue.

Ditto so if you really want to do the change here
I would suggest you try to test all use cases locally first.
We will need to track all in the description to ensure that all still working